src/components/GestureComponents.tsx

```diff
@@ -70,14 +112,35 @@ export const FlatList = React.forwardRef((props, ref) => {
       flatListProps[propName] = value;
     }
   }
+
   return (
     // @ts-ignore - this function cannot have generic type so we have to ignore this error
     <RNFlatList
       ref={ref}
       {...flatListProps}
       renderScrollComponent={(scrollProps) => (
-        <ScrollView {...{ ...scrollProps, ...scrollViewProps }} />
+        <ScrollView
+          {...{
+            ...scrollProps,
+            ...scrollViewProps,
+            waitFor: [
+              ...toArray(waitFor ? waitFor : []),
```

Comment:
What about `waitFor ?? []`?

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2137/commits/5571228feea976af1751e93422f8e8d82f30b548